src/components/shared/ButtonWithPanel/ArrowPanel.js

```diff
@@ -49,7 +50,11 @@ export class ArrowPanel extends React.PureComponent<Props, State> {
       }
       const openGeneration = state.openGeneration + 1;
 
-      setTimeout(this._onCloseAnimationFinish(openGeneration), 400);
+      clearTimeout(this.closeTimeout);
```

Comment:
hm, do we need to call this `clearTimeout` here?

Reply:
This is to prevent that we have 2 timeouts running together. This is probably rare but this does no harm IMO.

I'm not 100% sure but this might also make it possible to remove the `openGeneration` state.